File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/monitoring/model/PipelineSummary.java
package com.study.webflux.rag.domain.monitoring.model;

import java.time.Duration;
import java.time.Instant;
import java.util.List;
import java.util.Map;

public record PipelineSummary(
	String pipelineId,
	PipelineStatus status,
	Instant startedAt,
	Instant finishedAt,
	Map<String, Object> attributes,
	List<StageSnapshot> stages,
	List<String> llmOutputs,
	Long firstResponseLatencyMillis,
	Long lastResponseLatencyMillis
) {

	public long durationMillis() {
		if (startedAt == null || finishedAt == null) {
			return -1L;
		}
		return Duration.between(startedAt, finishedAt).toMillis();
	}
}

File: domain/src/main/kotlin/com/miyou/app/domain/monitoring/model/PipelineSummary.kt
package com.miyou.app.domain.monitoring.model

import java.time.Duration
import java.time.Instant

data class PipelineSummary(
    val pipelineId: String,
    val status: PipelineStatus,
    val startedAt: Instant?,
    val finishedAt: Instant?,
    val attributes: Map<String, Any?>,
    val stages: List<StageSnapshot>,
    val llmOutputs: List<String>,
    val firstResponseLatencyMillis: Long?,
    val lastResponseLatencyMillis: Long?,
) {
    fun durationMillis(): Long =
        if (startedAt == null || finishedAt == null) {
            -1L
        } else {
            Duration.between(startedAt, finishedAt).toMillis()
        }
}
